src/utils/src/Thread.c

```diff
@@ -78,7 +78,23 @@ PUBLIC_API STATUS defaultJoinThread(TID threadId, PVOID* retVal)
 
 PUBLIC_API VOID defaultThreadSleep(UINT64 time)
 {
-    Sleep((UINT32) (time / HUNDREDS_OF_NANOS_IN_A_MILLISECOND));
+    // Time in milliseconds
+    UINT64 remaining_time = time / HUNDREDS_OF_NANOS_IN_A_MILLISECOND;
+    // The loop will be run till the complete Sleep() time is reached
+    while(remaining_time > 0) {
```

Comment:
For unsigned integers it’s more prudent to check against not being 0 only

Reply:
Done